src/sage/schemes/elliptic_curves/ell_point.py

```diff
@@ -539,6 +539,47 @@ def __bool__(self):
         """
         return bool(self[2])
 
+    def has_order(self, n):
+        r"""
+        Test if this point has order exactly `n`.
+
+        INPUT:
+
+        - `n` -- integer, or its :class:`~sage.structure.factorization.Factorization`
+
+        ALGORITHM:
+
+        Compare a cached order if available, otherwise use :func:`sage.groups.generic.has_order`.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve('26b1')
+            sage: P = E(1, 0)
+            sage: P.has_order(7)
+            True
+            sage: P._order
+            7
```

Comment:
Can you add a test showing the cache is working (mainly calling it a second time)?

Can we also add the analogous tests for a factorization?

Reply:
Thanks, done.

Reply:
No, that actually doesn't test it, that "shows" the user that it is working. It needs to actually be called twice. Also, all of the factorization tests are `# not tested`.

Reply:
Added another (`tested`) test for `Factorization` objects. But I don't understand your other comment: The test checks that the `._order` attribute is set correctly after the call to `.has_order()`. How is this not sufficient?

Reply:
Ah, sorry, I missed that. Thank you.

Reply:
Actually, sorry, that isn’t correct. It isn’t that the attribute is set correctly but the code in the method that utilizes it is being run correctly. So the method needs to be called a second time after `_order` is set.

Reply:
Okay, I see. It should be done now.